components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go

```diff
@@ -118,6 +120,67 @@ type ResourceConfiguration struct {
 	Function *ResourceRequirements `json:"function,omitempty"`
 }
 
+type PodSpecTemplate struct {
+	// Compute Resources required by container created by function controller.
+	// +optional
+	Resources *ResourceRequirements `json:"resources,omitempty"`
+
+	// List of environment variables to set in the container.
+	// Cannot be updated.
+	// +optional
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	Env []v1.EnvVar `json:"env,omitempty"`
+
+	// Pod volumes to mount into the container's filesystem.
+	// +optional
+	// +patchMergeKey=mountPath
+	// +patchStrategy=merge
+	VolumeMounts []v1.VolumeMount `json:"volumeMounts,omitempty"`
+}
+
+type MetadataTemplate struct {
+	// Additional labels
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Additional annotations
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+type PodTemplate struct {
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	Metadata *MetadataTemplate `json:"metadata,omitempty"`
+
+	// Additional specification for pod created by function-controller
+	//+optional
+	Spec *PodSpecTemplate `json:"spec,omitempty"`
+}
+
+type Templates struct {
+	// Standard object's metadata.
+	// This metadata is applied on functionPod and buildPod.
+	// Can be overwritten by specific pod metadata
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	Metadata *MetadataTemplate `json:"metadata,omitempty"`
+
+	// Additional specification for build job pod
+	// +optional
+	BuildJobTemplate *PodTemplate `json:"buildJobTemplate,omitempty"`
+
+	// Additional specification for function pod
+	// +optional
+	FunctionPodTemplate *PodTemplate `json:"functionPodTemplate,omitempty"`
+
+	// Configuration of available volumes for function pod and buildjob pod
+	// +optional
+	Volumes []v1.Volume `json:"volumes,omitempty"`
```

Comment:
I would suggest moving this field to the `PodTemplate` struct. If we don't do this, we will lead to a possible situation when the volume mounted to one container would be unused in the second one

Reply:
Same topic: https://github.com/kyma-project/kyma/pull/15592/files#r985779218

Reply:
Moved.